### PR TITLE
docs: Fix outdated links

### LINKED ISSUE #2630: fix outdated links which were checked out by workflow/docs-url-alive-check
refer to https://github.com/kata-containers/kata-containers/pull/2482#issuecomment-918928643

We need to fix outdateed links
```sh
ERROR: Invalid URL 'https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/filesystems/dax.txt' found in the following files:

docs/design/architecture.md
ERROR: Invalid URL 'https://projectacrn.github.io/latest/tutorials/kbl-nuc-sdc.html#use-the-script-to-set-up-acrn-automatically' found in the following files:

docs/how-to/how-to-use-kata-containers-with-acrn.md
```
To
```sh
https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/filesystems/dax.rst

https://projectacrn.github.io/2.1/tutorials/kbl-nuc-sdc.html#use-the-script-to-set-up-acrn-automatically
```

### PR DESCRIPTION
**Pull request by Bevisy**
fix outdated links which were checked out by workflow/docs-url-alive-check

Fixes #2630

Signed-off-by: Binbin Zhang <binbin36520@gmail.com>

### CODE REVIEW

File: docs/how-to/how-to-use-kata-containers-with-acrn.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ This document requires the presence of the ACRN hypervisor and Kata Containers o
 
 - ACRN supported [Hardware](https://projectacrn.github.io/latest/hardware.html#supported-hardware).
   > **Note:** Please make sure to have a minimum of 4 logical processors (HT) or cores.
-- ACRN [software](https://projectacrn.github.io/latest/tutorials/kbl-nuc-sdc.html#use-the-script-to-set-up-acrn-automatically) setup.
+- ACRN [software](https://projectacrn.github.io/2.1/tutorials/kbl-nuc-sdc.html#use-the-script-to-set-up-acrn-automatically) setup.
```
Comment (fidencio) [inline docs/how-to/how-to-use-kata-containers-with-acrn.md:25]: @gvancuts, do you happen to know why this got removed from the `latest` and what would be a good replacement for that URL?

@Bevisy, I think we should not point to the `2.1` repo, but instead check whether something has to be updated from our side, and Geoffrey may be the best person to guide us through the needed changes.

**Review by fidencio — COMMENTED** [review]
I'd put this on hold till we learn about the best approach from @gvancuts.

File: docs/how-to/how-to-use-kata-containers-with-acrn.md (hunk 1 above)
Comment (amshinde) [reply docs/how-to/how-to-use-kata-containers-with-acrn.md:25]: cc @vijaydhanraj Do you know why those doc pages were removed for releases following 2.1? Can you suggest what would be the appropriate doc to link to here ?

**Comment by amshinde** [comment]
> I agree with @vijaydhanraj . As for this pr, maybe it only needs to include DAX-related modifications

@Bevisy Please include the change related to ACRN as well for the CI to pass. Update to the new link (https://projectacrn.github.io/latest/tutorials/run_kata_containers.html for the CI to) as pointed out by @vijaydhanraj

**Review by GabyCT — APPROVED** [review]

**Pull request merged**